src/lib/getDataAndInitListeners.js

```diff
@@ -0,0 +1,169 @@
+
+  contract.on('StakeBurned', () => updateProductsByStakingPool(poolId));
+  contract.on('DepositExtended', () => updateProductsByStakingPool(poolId));
+  contract.on('StakeDeposited', () => updateProductsByStakingPool(poolId));
+  contract.on('PoolFeeChanged', () => updateProductsByStakingPool(poolId));
+  contract.on('StakeBurned', () => updateProductsByStakingPool(poolId));
+
+  stakingPoolContracts.push(contract);
+}
+
+async function subscribeToAllStakingPoolDependantEvents() {
+  const stakingPools = await fetchStakingPoolIdAndAddress();
+
+  for (const pool of stakingPools) {
+    const { address, id } = pool;
+    subscribeToStakingPoolDependantEvents(id, address);
+  }
+}
+
+async function subscribeToPoolAllocationChanges() {
+  const stakingPools = await fetchStakingPoolIdAndAddress();
+
+  for (const pool of stakingPools) {
+    const { address, id } = pool;
+    subscribeToStakingPoolDependantEvents(id, address);
+  }
+}
+
+async function subscribeToCoverEvents() {
+  const Cover = new ethers.Contract(CONTRACTS_ADDRESSES.Cover, CoverAbi, wsProvider);
+
+  Cover.on('ProductSet', productId => fetchProductDataById(productId));
+  Cover.on('CoverEdited', (coverId, productId) => fetchProductDataById(productId));
+
+  // TODO: add global capacity factor listener
+}
+
+async function subscribeToNewStakingPools() {
+  const StakingPoolFactory = new ethers.Contract(
+    CONTRACTS_ADDRESSES.StakingPoolFactory,
+    StakingPoolFactoryAbi,
+    wsProvider,
+  );
+
+  StakingPoolFactory.on('StakingPoolCreated', async (id, address) => {
+    await fetchStakingPoolData(id);
+    const productsData = await fetchAllProductDataForPool(id);
+    for (const product of productsData) {
+      const { productId, trancheCapacities, blockNumber } = product;
+      store.dispatch({
+        type: actions.SET_PRODUCT_STAKING_POOL,
+        payload: {
+          productId,
+          poolId: id,
+          stakingPoolData: {
+            trancheCapacities,
+            blockNumber,
+          },
+        },
+      });
+    }
+    subscribeToStakingPoolDependantEvents(id, address);
+  });
+}
+
+module.exports = async function (app) {
+  store = app.get('store');
+  if (!store) {
+    throw Error('Store not initialized');
+  }
+
+  await fetchInitialData();
+
+  // tranche expiration checker
+  const trancheChecker = setInterval(async () => {
+    const activeTrancheId = calculateCurrentTrancheId();
+    if (activeTrancheId === trancheId) {
+      await fetchAllProductsData();
+    }
+  }, 10000);
+
+  // subscribe to Pool events
+  await subscribeToAllStakingPoolDependantEvents();
+  await subscribeToNewStakingPools();
+
+  await subscribeToPoolAllocationChanges();
+
+  // subscribe to Cover Events
+  await subscribeToCoverEvents();
+
+  process.on('SIGTERM', () => {
+    for (const contract of stakingPoolContracts) {
+      contract.removeAllListeners();
+    }
+    clearInterval(trancheChecker);
+  });
```

Comment:
This is redundant. If the process is terminating everything's killed anyway. Is anything preventing the process from shutting down if you don't do this?

Reply:
Added graceful shutdown if we need to control this in the future